libfreerdp/utils/cliprdr_utils.c

```diff
@@ -131,6 +111,66 @@ UINT cliprdr_parse_file_list(const BYTE* format_data, UINT32 format_data_length,
 	return result;
 }
 
+BOOL cliprdr_read_filedescriptor(wStream* s, FILEDESCRIPTORW* file)
+{
+	UINT64 tmp = 0;
+	WINPR_ASSERT(file);
+
+	if (!Stream_CheckAndLogRequiredLength(TAG, s, sizeof(FILEDESCRIPTORW)))
+		return FALSE;
+
+	Stream_Read_UINT32(s, file->dwFlags); /* flags (4 bytes) */
+	Stream_Read_UINT32(s, file->clsid.Data1);
+	Stream_Read_UINT16(s, file->clsid.Data2);
+	Stream_Read_UINT16(s, file->clsid.Data3);
+	Stream_Read(s, &file->clsid.Data4, sizeof(file->clsid.Data4));
+	Stream_Read_INT32(s, file->sizel.cx);
+	Stream_Read_INT32(s, file->sizel.cy);
+	Stream_Read_INT32(s, file->pointl.x);
+	Stream_Read_INT32(s, file->pointl.y);
+	Stream_Read_UINT32(s, file->dwFileAttributes); /* fileAttributes (4 bytes) */
+	Stream_Read_UINT64(s, tmp);                    /* ftCreationTime (8 bytes) */
+	file->ftCreationTime = uint64_to_filetime(tmp);
+	Stream_Read_UINT64(s, tmp); /* ftLastAccessTime (8 bytes) */
+	file->ftLastAccessTime = uint64_to_filetime(tmp);
+	Stream_Read_UINT64(s, tmp); /* lastWriteTime (8 bytes) */
+	file->ftLastWriteTime = uint64_to_filetime(tmp);
+	Stream_Read_UINT32(s, file->nFileSizeHigh); /* fileSizeHigh (4 bytes) */
+	Stream_Read_UINT32(s, file->nFileSizeLow);  /* fileSizeLow (4 bytes) */
+	Stream_Read_UTF16_String(s, file->cFileName,
+	                         ARRAYSIZE(file->cFileName)); /* cFileName (520 bytes) */
+	return TRUE;
+}
+
+BOOL cliprdr_write_filedescriptor(wStream* s, const FILEDESCRIPTORW* file)
+{
+	WINPR_ASSERT(file);
+
+	if (!Stream_EnsureRemainingCapacity(s, sizeof(FILEDESCRIPTORW)))
+		return FALSE;
+
+	Stream_Write_UINT32(s, file->dwFlags); /* flags (4 bytes) */
+
+	Stream_Write_UINT32(s, file->clsid.Data1);
+	Stream_Write_UINT16(s, file->clsid.Data2);
+	Stream_Write_UINT16(s, file->clsid.Data3);
+	WINPR_ASSERT(sizeof(file->clsid.Data4) == 8);
```

Comment:
strange assert !

Reply:
indeed, that one was a copy & paste one, so will update.